### ISSUE TITLE
Several contact tests are forcing PETSc to malloc new memory

### ISSUE THREAD

**Issue by permcody** [idaholab/moose#4690]
There are several tests that fail in `modules/combined` with the following option:
```Problem/error_on_jacobian_nonzero_reallocation=true```

If we are reallocing then we are doing something wrong. This should be fixed. To trigger these errors through the test harness, do this:
```./run_tests --cli-args Problem/error_on_jacobian_nonzero_reallocation=true```

**Comment by permcody** [comment]
@friedmud - pinging your for a little insight here... I believe the issue is that we aren't doing the initial geometry search in the right place. Our non-zero pattern is not correct when doing the first solve.

**Comment by friedmud** [comment]
Yes: that's what is happening.  I actually suspect that we don't know about the geometric search objects soon enough in the initialization cycle.  We may need to reinit the system again before we start the solve so we can fix this... But it needs research.

**Comment by permcody** [comment]
@fdkong - Is this ticket still relevant? It's old and we may have fixed this by now.
